public/upload.js

```diff
@@ -0,0 +1,147 @@
+
+          var fd = new FormData();
+          fd.append("fname", file.name);
+          fd.append("data", blob, file.name);
+
+          var xhr = new XMLHttpRequest();
+          var hex = ivToStr(random_iv);
+          xhr.open("post", "/upload/" + hex, true);
+
+          var row = document.createElement("tr");
+          var name = document.createElement("td");
+          var link = document.createElement("td");
+          var expiry = document.createElement("td");
+
+          var cellText = document.createTextNode(file.name);
+          //name.innerHTML = file.name;
+          name.appendChild(cellText);
+
+          var progress = document.createElement("p");
+          //li.appendChild(progress);
+          row.appendChild(name);
+          row.appendChild(link);
+          row.appendChild(expiry);
+          document.getElementById("uploaded-files").appendChild(row);
+
+
+          xhr.upload.addEventListener("progress", returnBindedLI(progress, name, link, row));
+
+          xhr.onreadystatechange = function() {
+            if (xhr.readyState == XMLHttpRequest.DONE) {
+              window.crypto.subtle.exportKey("jwk", key).then(function(keydata) {
+                var curr_name = localStorage.getItem(file.name);
+
+                localStorage.setItem(hex, xhr.responseText);
+
+                link.innerHTML = "http://localhost:3000/download/" + hex + "/#" + keydata.k;
+                link.setAttribute("href", "http://localhost:3000/download/" + hex + "/#" + keydata.k);
+
+                console.log("Share this link with a friend: http://localhost:3000/download/" + hex + "/#" + keydata.k);
+              })
+            }
+          };
+
+          xhr.send(fd);
+        })
+        .catch(function(err){
+          console.error(err);
+        });
+
+      })
+      .catch(function(err){
+        console.error(err);
+      });
+
+    };
+    reader.readAsArrayBuffer(file);
+  }
+
+  function ivToStr(iv) {
+    let hexStr = "";
+    for (var i in iv) {
+      if (iv[i] < 16) {
+        hexStr += "0" + iv[i].toString(16);
+      } else {
+        hexStr += iv[i].toString(16);
+      }
+    }
+    window.hexStr = hexStr;
+    return hexStr;
+  }
+
+  function strToIv(str) {
+    var iv = new Uint8Array(16);
+    for (var i = 0; i < str.length; i += 2) {
+      iv[i/2] = parseInt((str.charAt(i) + str.charAt(i + 1)), 16);
+    }
+
+    return iv;
+  }
+
+  function returnBindedLI(a_element, name, link, li) {
+    return function updateProgress(e) {
+      if (e.lengthComputable) {
+        var percentComplete = Math.floor((e.loaded / e.total) * 100);
+        a_element.innerHTML = "Progress: " + percentComplete + "%";
+
+        if (percentComplete === 100) {
+          var del = document.createElement("td");
+          var btn = document.createElement("button");
+          btn.innerHTML = "x";
+          btn.style = "padding: 0; border: none; background: none; cursor: pointer"
```

Comment:
You can add a class to the button when the status changes, then style that class from the css file.